contracts/AllPayAuction.sol

```diff
@@ -110,15 +114,21 @@ contract AllPayAuction is Auction {
         auction.availableFunds = 0;
         uint256 fees = (auction.protocolFee * withdrawAmount) / 10000;
         address feeRecipient = protocolParameters.treasury();
-        sendFunds(false, auction.biddingToken, auction.auctioneer, withdrawAmount - fees);
-        sendFunds(false, auction.biddingToken,feeRecipient,fees);
+        sendERC20(auction.biddingToken, auction.auctioneer, withdrawAmount - fees);
+        sendERC20(auction.biddingToken,feeRecipient,fees);
         emit Withdrawn(auctionId, withdrawAmount);
     }
 
     function claim(uint256 auctionId) external exists(auctionId) onlyAfterDeadline(auctions[auctionId].deadline) notClaimed(auctions[auctionId].isClaimed) {
         AuctionData storage auction = auctions[auctionId];
         auction.isClaimed = true;
-        sendFunds(auction.auctionType == AuctionType.NFT, auction.auctionedToken, auction.winner, auction.auctionedTokenIdOrAmount);
+        if(auction.auctionType == AuctionType.NFT) {
```

Comment:
Now, this is bad, because now this conditional code smell leaked to all the auction contracts.

Ultimately, this problem is unsolvable in an elegant way, because it stems from fundamental incompatibilities between ERC20 and ERC721 and limitations of subtyping in Solidity.

But I think we can still improve this by doing the following in Auction.sol:

* Keep `sendERC20`, `sendNFT`, `receiveERC20`, `receiveNFT` in auction.sol and keep using them in the "good" place where you know which function to use.

* But have also `send` and `receive` functions in auction.sol and use them in the "bad" cases like this, where you don't know which function to call. The `send` and `receive` functions will then do the ugly conditionals and decide which function to call.